### PR TITLE
add stream support

### PR DESCRIPTION
**Pull request by zertosh**
Adds stream support to gzip-size so I can then add stream support to [gulp-size](https://github.com/sindresorhus/gulp-size). cc: @sindresorhus

### CODE REVIEW

**Comment by sindresorhus** [comment]
By concating the stream you somewhat lose part of the benefits of streams. You get streamed gzipping, which is good, but in a real stream gulp chain this would make streaming moot as the streaming stops on this step. Instead we could maybe expose a gzipSize stream that emitted a `size` event when available, or just attached the size onto the stream and could be fetched in the `end` event. Thoughts?

**Comment by zertosh** [comment]
Yeah totally – what do you think now?

File: index.js [hunk 1]
```diff
@@ -17,6 +19,21 @@ module.exports = function (str, cb) {
...
+			size += buf.length;
+		})
+		.on('error', function () {
+			this.size = 0;
+		})
+		.on('end', function () {
+			this.size = size;
+			this.emit('size', size);
+		});
+};
```
Comment (sindresorhus) [inline index.js]: Actually, I don't think it makes sense to return the gzip stream. The user only cares about the gzipped size. They don't actually want their stream gzipped. Rather return a passthrough stream with the event and property. I would also make the event name `gzip-size` and property `gzipSize` just to be explicit.

Comment (zertosh) [inline index.js]: Done! I noticed that [ava](https://www.npmjs.com/package/ava) wasn't correctly counting the test plan. I tried to go to the repo but it 404s. I'm guessing you discontinued it? Anyway, I switched to [tap](https://github.com/isaacs/node-tap) – hope that's ok.

Comment (sindresorhus) [inline index.js]: Ava isn't discontinued, just never published, and I haven't had time to open source it yet. Using tap is fine.

**Comment by sindresorhus** [comment]
Excellent! Exactly what I was thinking :)

**Pull request merged**